libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -430,3 +421,32 @@ std::optional<YulString> DataFlowAnalyzer::isSimpleLoad(
 				return key->name;
 	return {};
 }
+
+bool DataFlowAnalyzer::hasFlowOutControlFlow(Block const& _block) const
+{
+	return
+		_block.statements.empty() ||
+		TerminationFinder{m_dialect, &m_controlFlowSideEffects}.
+			controlFlowKind(_block.statements.back()) == TerminationFinder::ControlFlow::FlowOut;
```

Comment:
I find this rather creepy... I'd much rather have ``TerminationFinder`` run on the full AST up front and map block pointers to out flow or something like that...
Every time I see this, I expect bugs due to ``{ break leave }`` or ``{ continue revert(0,0) }`` or ``{ if someCond { continue } revert(0,0) }`` or something...
Although right now I'm not finding a case that will *actually* break...

Do we have tests for these things? Maybe we should have.

But maybe I'm paranoid here... we do remove unreachable stuff like that anyways and the for loop handling in here should still remain sound...

Reply:
But this function is totally fine even if we didn't remove dead code already, isn't it?

1) This function returning "true" can always be wrong because of some complicated invariants / conditions
2) If the last statement in the block is a non-flowout statement, but there is additional control-flow, then we must handle that correctly already anyway (continue / break)

Note that `leave` is always ignored in this component.